src/pybind/mgr/cli_api/tests/test_cli_api.py

```diff
@@ -0,0 +1,37 @@
+import unittest
+
+from ..module import CLI, BenchmarkException, HandleCommandResult
+
+
+class BenchmarkRunnerTest(unittest.TestCase):
+    def test_number_of_calls_on_start_fails(self):
+        with self.assertRaises(BenchmarkException) as ctx:
+            CLI('CLI', 0, 0).benchmark(0, 10, 'list_servers', [])
+        self.assertEqual(str(ctx.exception),
+                         "Number of calls and number "
+                         "of parallel calls must be greater than 0")
+
+    def test_number_of_parallel_calls_on_start_fails(self):
+        with self.assertRaises(BenchmarkException) as ctx:
+            CLI('CLI', 0, 0).benchmark(100, 0, 'list_servers', [])
+        self.assertEqual(str(ctx.exception),
+                         "Number of calls and number "
+                         "of parallel calls must be greater than 0")
+
+    def test_number_of_parallel_calls_on_start_works(self):
+        CLI.benchmark(10, 10, "get", "osd_map")
+
+    def test_function_name_fails(self):
+        for iterations in [0, 1]:
+            threads = 0 if iterations else 1
+            with self.assertRaises(BenchmarkException) as ctx:
+                CLI('CLI', 0, 0).benchmark(iterations, threads, 'fake_method', [])
```

Comment:
It's a bit hard to read this loop I would suggest something like:
```suggestion
    def test_function_name_fails(self):
        def t(threads, iterations):
            with self.assertRaises(BenchmarkException) as ctx:
                CLI('CLI', 0, 0).benchmark(iterations, threads, 'fake_method', [])
         t(0, 1)
         t(1, 0)
```
Moreover, I think if you're testing that the function is an existing one you should test this part of the code:
```
        if iterations and threads:
            try:
                func = getattr(self, func_name)
            except AttributeError:
                return HandleCommandResult(errno.EINVAL,
                                           stderr="Could not find the public "
                                           "function you are requesting")
```
Not this one:

```
else:
    raise BenchmarkException("Number of calls and number "
         "of parallel calls must be greater than 0")
  ```